### PR TITLE
NIFI-11303 Added a go-to entry on the right click context menu for th…

### PR DESCRIPTION
**Pull request by Freedom9339**
…e provenance lineage tree to take you to the specified component in the graph

# Summary

[NIFI-11303](https://issues.apache.org/jira/browse/NIFI-11303) - Added a Go To entry to the context menu when right clicking an element in the provenance lineage tree. When clicked, it will take you to the component that was right clicked on in the graph. This will make it more convenient for the user as they will no longer have to copy the UUID of the component and search for it in the graph.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by markobean** [comment]
I kicked the tires on this PR. The update looks pretty straight-forward, but I would like someone with some more NiFi UI experience to weigh in on the code changes. I can say this much in support of this PR. I tested it out by installing and running NiFi. Then, I checked out the provenance options targeted here. Everythinng worked just fine. The provenance lineage graph now has a "go to" option on each node (right-click) which correctly takes you to the component in the graph which generated the event. 

Great job! Looking forward to this feature.

**Review by mtien-apache — CHANGES_REQUESTED** [review]
@Freedom9339 Thanks for the PR! The new menu item works well. I only have one comment below. Once that is addressed, it's a +1 from me.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js [hunk 1]
```diff
@@ -1213,6 +1195,9 @@
                     }
                 };
 
+                // used to access provenance data when outside of scope
+                var provenanceData;
+
```
[the remark this answers is not in the document]
Comment (Freedom9339) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js:1200]: @mtien-apache Thank you for the review! This variable is used here:

https://github.com/apache/nifi/blob/7012bfe58da3a5c131896e09fef6888ed300e736/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js#L707

And it is being set so we can call getItemByID() in the goTo() function:

https://github.com/apache/nifi/blob/7012bfe58da3a5c131896e09fef6888ed300e736/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js#L1238-L1243

[the remark this answers is not in the document]
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js:1200]: @Freedom9339 I see `provenanceData` is referenced from the line above:
https://github.com/apache/nifi/blob/7012bfe58da3a5c131896e09fef6888ed300e736/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/provenance/nf-provenance-table.js#L704-L706

When I remove the variable from line 1199, the functionality still works as expected.

**Review by mtien-apache — APPROVED** [review]
@Freedom9339 LGTM. Thanks for the fix!

**Pull request merged**